stack_test.go

```diff
@@ -0,0 +1,198 @@
+			"orgName": "MyOrg",
+			"name": "mystack.grafana.net",
+			"url": "https://mystack.grafana.net",
+			"slug": "mystack",
+			"version": "stable",
+			"description": "My amazing stack",
+			"status": "active",
+			"gateway": "istio",
+			"createdAt": "2021-12-22T14:02:46.000Z",
+			"createdBy": "xyz",
+			"updatedAt": null,
+			"updatedBy": "",
+			"trial": 0,
+			"trialExpiresAt": null,
+			"clusterId": 666,
+			"clusterSlug": "prod-eu-west-0",
+			"clusterName": "prod-eu-west-0",
+			"plan": "gcloud",
+			"planName": "Grafana Cloud",
+			"billingStartDate": "2021-12-22T14:02:46.000Z",
+			"billingEndDate": null,
+			"billingActiveUsers": 0,
+			"currentActiveUsers": 1,
+			"currentActiveAdminUsers": 1,
+			"currentActiveEditorUsers": 0,
+			"currentActiveViewerUsers": 0,
+			"dailyUserCnt": 0,
+			"dailyAdminCnt": 0,
+			"dailyEditorCnt": 0,
+			"dailyViewerCnt": 0,
+			"billableUserCnt": 1,
+			"dashboardCnt": 6,
+			"datasourceCnts": {},
+			"userQuota": 10,
+			"dashboardQuota": -1,
+			"alertQuota": -1,
+			"ssl": true,
+			"customAuth": true,
+			"customDomain": true,
+			"support": true,
+			"runningVersion": "8.3.3 (commit: 30bb7a93c, branch: HEAD)",
+			"machineLearning": 0,
+			"hmInstancePromId": 112233,
+			"hmInstancePromUrl": "https://prometheus-prod-01-eu-west-0.grafana.net",
+			"hmInstancePromName": "mystack-prom",
+			"hmInstancePromStatus": "active",
+			"hmInstancePromCurrentUsage": 11111,
+			"hmInstancePromCurrentActiveSeries": 222222,
+			"hmInstanceGraphiteId": 111111,
+			"hmInstanceGraphiteUrl": "https://graphite-prod-01-eu-west-0.grafana.net",
+			"hmInstanceGraphiteName": "mystack-graphite",
+			"hmInstanceGraphiteType": "graphite-v5",
+			"hmInstanceGraphiteStatus": "active",
+			"hmInstanceGraphiteCurrentUsage": 0,
+			"hlInstanceId": 121416,
+			"hlInstanceUrl": "https://logs-prod-eu-west-0.grafana.net",
+			"hlInstanceName": "mystack-logs",
+			"hlInstanceStatus": "active",
+			"hlInstanceCurrentUsage": 0,
+			"amInstanceId": 654321,
+			"amInstanceName": "mystack-alerts",
+			"amInstanceUrl": "https://alertmanager-eu-west-0.grafana.net",
+			"amInstanceStatus": "active",
+			"amInstanceGeneratorUrl": "https://mystack.grafana.net",
+			"htInstanceId": 123456,
+			"htInstanceUrl": "https://tempo-eu-west-0.grafana.net",
+			"htInstanceName": "mystack-traces",
+			"htInstanceStatus": "active",
+			"regionId": 3,
+			"regionSlug": "eu"
+			}
+	]
+}
+`
+)
+
+func TestStacks(t *testing.T) {
+	server, client := gapiTestTools(t, 200, getStacksJSON)
+	defer server.Close()
+
+	stacks, err := client.Stacks()
+
+	if err != nil {
+		t.Fatalf("expected error to be nil; got: %s", err.Error())
+	}
+
+	actualItemCound := len(stacks.Items)
+	expectedItemCount := 1
+
+	if actualItemCound != expectedItemCount {
+		t.Errorf("Length of returned stacks - Actual Stacks Count: %d, Expected Stacks Count: %d", actualItemCound, expectedItemCount)
+	}
+
+	actualStackID := stacks.Items[0].ID
+	expectedStackID := int64(1)
```

Comment:
that field/value is already present in data you provided, in `getStacksJSON`, why we duplicate it? 
could we use something like `reflect.DeepEqual` to compare expected with actual data without comparing field by field "manually"?

Reply:
Yes I have used different json to keep the concerns of the tests separate. `getStacksJson` contains a collection stacks. The others are just one `stack`.

Reply:
Regarding deep Equal, I am not sure we understand the API enough to do that kind of matching.